packages/macro/src/macroJs.ts

```diff
@@ -111,6 +111,34 @@ export default class MacroJs {
       return false
     }
 
+    // t(`id`)`message` -> i18n._({ id: "id", message: "message" })
+    if (
+      this.types.isCallExpression(path.node) &&
+      this.types.isTaggedTemplateExpression(path.parent) &&
+      (this.types.isStringLiteral(path.node.arguments[0]) ||
+        this.types.isTemplateLiteral(path.node.arguments[0])) &&
+      this.isLinguiIdentifier(path.node.callee, "t")
+    ) {
+      const idTokens = this.types.isTemplateLiteral(path.node.arguments[0])
+        ? this.tokenizeTemplateLiteral(path.node.arguments[0])
+        : [
+            {
+              type: "text",
+              value: this.expressionToArgument(path.node.arguments[0]),
+            } as TextToken,
+          ]
```

Comment:
You don't need to parse id node into tokens, tokens are used only for message. You need to just check if it's string literal or tpl literal and pass this node directly to message descriptor factory.